prow_config.yaml

```diff
@@ -28,59 +28,32 @@ workflows:
     params:
       installIstio: true
       workflowName: deployapp-istio
+  ### GCP IAP
   - py_func: kubeflow.kubeflow.ci.kfctl_e2e_workflow.create_workflow
     name: kfctl-go-iap-endpoint
     job_types:
+      - presubmit
       - postsubmit
       - periodic
     include_dirs:
       - bootstrap/*
-      - deployment/*
-      - dependencies/*
       - kubeflow/*
       - testing/*
+      - py/kubeflow/kubeflow/ci/kfctl_e2e_workflow.py
     kwargs:
       use_basic_auth: false
       test_endpoint: true
       config_path: https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_gcp_iap.yaml
-  - py_func: kubeflow.kubeflow.ci.kfctl_e2e_workflow.create_workflow
-    component: kfctl_go_test
-    name: kfctl-go-iap-istio
-    job_types:
-      - presubmit
-    include_dirs:
-      - bootstrap/*
-      - kubeflow/*
-      - testing/*
-    kwargs:
-      use_basic_auth: false
-      config_path: https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_gcp_iap.yaml
-  # If the kfctl_e2e_workflow.py was modified then on presubmit we also want to run with test_endpoint true
-  # to ensure that test still pasess
-  - py_func: kubeflow.kubeflow.ci.kfctl_e2e_workflow.create_workflow
-    component: kfctl_go_test
-    name: kfctl-gcp-e2e-changed
-    job_types:
-      - presubmit
-    include_dirs:
-      - bootstrap/*
-      - kubeflow/*
-      - testing/*
-      - py/kubeflow/kubeflow/ci/kfctl_e2e_workflow.py 
-    kwargs:
-      use_basic_auth: false
-      test_endpoint: true
-      config_path: https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_gcp_iap.yaml
-  # Run basic auth test as part of every periodic and postsubmit run. 
+  ### GCP BasicAuth
   - py_func: kubeflow.kubeflow.ci.kfctl_e2e_workflow.create_workflow
     name: kfctl-go-basic-auth
     job_types:
+      - presubmit
```

Comment:
I would actually suggest we just run basic auth in postsubmit.

Per kubeflow/kfctl#57 all the expensive presubmits are slowing us down. Using an E2E test for Kubeflow as a whole to test that kfctl is something we should fix.

At best we should only be triggering this if the gcp/basic auth code changes.